### PR TITLE
Prevent change of contact-color when changing a contacts adressbook, fix #243

### LINKED ISSUE #243: Prevent change of contact-color when changing a contacts adressbook

### PR DESCRIPTION
**Pull request by skjnldsv**
Prevent change of contact-color when changing a contacts adressbook, fix #243

### CODE REVIEW

**Comment by jancborchardt** [comment]
@skjnldsv as pull request title and commit message, please always describe what was done. :) You can add the `fix issuenumber` at the end so it will be referenced, but it should always be human-readable / understandable.

**Comment by jancborchardt** [comment]
@aspdye please review this as you opened the original issue :)

File: js/services/contact_service.js [hunk 1]
```diff
@@ -76,10 +76,13 @@ angular.module('contactsApp')
 		}
 	};
 
-	this.create = function(newContact, addressBook) {
+	this.create = function(newContact, addressBook, uid) {
 		addressBook = addressBook || AddressBookService.getDefaultAddressBook();
 		newContact = newContact || new Contact(addressBook);
 		var newUid = uuid4.generate();
```
Comment (Henni) [inline js/services/contact_service.js]: Please put this into an else clause. We don't have to do this if we have a valid uid.

**Comment by skjnldsv** [comment]
Indeed, we could save a few resources!
I did it at first, but ended up defining it before because the tests failed and didn't allowed me to define twice Newuuid with var :)

**Comment by skjnldsv** [comment]
Rebased!
There's a little flash when changing a contact addressbook, the delay between the contact deletion and the new contact creation.
The way we do this isn't maybe the right way, or could we stop the digest of angular between the delete and the new contact? I don't know. :/

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by skjnldsv** [comment]
Thanks @DeepDiver1975

**Pull request merged**